src/components/main-view/main-view.jsx

```diff
@@ -2,63 +2,117 @@ import React, { useState, useEffect } from 'react';
 import { MovieCard } from "../movie-card/movie-card";
 import { MovieView } from "../movie-view/movie-view";
 import { LoginPage } from "../login-view/login-page";
-import "./main-view.css"
+import { SignUp } from '../signup-view/signup-view';
+import "./main-view.css";
+import { Button, Row, Col } from "react-bootstrap";
 
 export const MainView = () => {
-  const [movies, setMovies] = useState([]);
 
+  const storedToken = localStorage.getItem("token");
+  const storedUser = JSON.parse(localStorage.getItem("user"));
+  const [movie, setMovie] = useState([]);
   // state changes for selected movies
   const [selectedMovie, setSelectedMovie] = useState(null);
-  const [user, setUser] = useState(null);
+  const [user, setUser] = useState(storedUser ? storedUser : null);
+  const [token, setToken] = useState(storedToken ? storedToken : null);
+
+  const chunkArray = (array, chunkSize) => {
+    let results = [];
+    while (array.length) {
+      results.push(array.splice(0, chunkSize));
+    }
+    return results;
+  };
+
+  const movieRows = chunkArray([...movie], Math.ceil(movie.length / 2));
```

Comment:
It's not clear to me what's the purpose of this function.

Reply:
I used it to make two rows into four columns each